.github/workflows/ci.yaml

```diff
@@ -111,13 +111,16 @@ jobs:
       - uses: actions/checkout@v4
       - name: "Install Rust toolchain"
         run: rustup show
-      - name: "Install cargo insta"
+      - name: "Install mold"
+        uses: rui314/setup-mold@v1
+      - name: "Install cargo nextest"
         uses: taiki-e/install-action@v2
         with:
-          tool: cargo-insta
+          tool: cargo-nextest
       - uses: Swatinem/rust-cache@v2
       - name: "Run tests"
-        run: cargo insta test --all --all-features --unreferenced reject
+        shell: bash
+        run: cargo nextest run --all --status-level skip --failure-output immediate-final --no-fail-fast -j 12
```

Comment:
Is `-j 12` correct for default size runners? The default runners have 2-4 cores (https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories), only the large runners have 12 (https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners#about-macos-larger-runners). CC @zanieb iirc you benchmarked this?

Reply:
Yeah I benchmarked this for our other project where I believe we have 8 cores. I found going above the number of cores was beneficial to a point (the default just uses the same number of jobs as you have cores).

I'd say 12 is probably too high over here with the small runner.

Reply:
I'll try a few sizes.